File: model-client/src/main/java/de/q60/mps/shadowmodels/runtime/model/INodeResolveContext.java
package de.q60.mps.shadowmodels.runtime.model;

import de.q60.mps.incremental.util.ContextValue;

public interface INodeResolveContext {
  ContextValue<INodeResolveContext> CONTEXT_VALUE = new ContextValue<>();
  INode resolve(INodeReference ref);
}

File: model-client/src/main/kotlin/org/modelix/model/api/INodeResolveContext.kt
package org.modelix.model.api

import org.modelix.model.util.ContextValue

interface INodeResolveContext {
    fun resolve(ref: INodeReference?): INode?

    companion object {
        @JvmField
        val CONTEXT_VALUE = ContextValue<INodeResolveContext>()
    }
}
